pkg/plugin/twinmaker/handler.go

```diff
@@ -470,6 +476,12 @@ func (s *twinMakerHandler) GetAlarms(ctx context.Context, query models.TwinMaker
 		}
 		failures = append(failures, newFailures...)
 		pValues = append(pValues, propertyReferences...)
+		if isLimited && len(pValues) >= maxResults {
```

Comment:
We need a more deep rooted check on the maxResults in the GetPropertyValueHistory client.
The `pValues` will have collected all the data even with maxResults passed into GetPropertyValueHistory since the field only limits the results per page. Even if maxResults is 1 our code will see a nextToken and make more API calls to collect all data.
We need to collect as many pages as satisfies the maxResults (250 max results per page) - so the client should be updated to stop requesting data when the max is reached.

Reply:
There is an issue that the following convoluted scenarios demonstrates:

We have 2 alarms: alarmA and alarmB 
maxResults = 2
max results per page = 3


Normally, we get:

1, alarmA  = blah
2, alarmA  = blah
3, alarmA  = blah

on page 1

and then 

4, alarmA  = blah
5, alarmB  = blah

on page 2.


In the current code, we go through all pages and get values for both alarmA and alarmB.
With the current code, we stop on page 1 as we have more than 2 results. But we miss alarmB's data point.

Reply:
We need to keep a modifiable set of alarms we need data for.

Once we have data for an alarm, we can remove it from the set so we are no longer checking for it.

Once either all the pages have been exhausted or there are no more alarms we need data for, then we can exit and return from the function.


We still need to optimise though. 
If alarmA has 1 million data points, and we get a hit on the first page, we don't want to carry on going through the following x pages which are all alarmA data points.

We could go through a page, remove all of the alarms we found data for, and then start the process again from page 1 but call the function with the new shorter list of alarms.

A bit convoluted I do admit.

Reply:
@msshui-amzn Not sure if I follow entirely. Would you be able to write this and push to this branch?

Reply:
@lean257 sure, i will make the change.

Reply:
apparently, you cannot do what I described.

I added an extra function where I only store the latest value. I see that you also do that anyway in the `getAlarms` function. But the new function i wrote isolates the call from the other getComponentHistory call and hopefully can be modified once what i have described is possible.